src/util/prolog/prologSession.ts

```diff
@@ -1,13 +1,65 @@
+import fs from "fs";
+import path from "path";
 import { format_answer, type } from "tau-prolog/modules/core";
-import {PrologResult} from "./prologResult";
+import {kahnAlgorithm, regexGroupToArray} from "../helperFunctions";
+import { PrologResult } from "./prologResult";
+import { removeFileExtension, removeModuleDeclarations, removeNonFunctionalities } from "./prologStringManipulation";
+
+// RegExp group one will be the file to be imported.
+const fetchPrologImportsRegExp = /:-(?: |)use_module\(([a-zA-Z1-9]*)\) *\./g;
+
+// relativePath is relative to this very file (prologSession.ts)
+export function importFile(relativePath: string): PrologSession {
+	const absolutePath = path.resolve(__dirname, relativePath);
+	const program = fs.readFileSync(absolutePath, "utf-8");
+	const fileName = removeFileExtension(path.basename(absolutePath));
+	return new PrologSession(program, path.dirname(absolutePath), fileName);
+}
+
+function resolveImports(program: string, defaultPath: string, currentFileName: string = "main"): string {
+		const dependencyGraph: Set<string[]> = getDependencyGraph(program, defaultPath, currentFileName);
+		const importList: string[] = kahnAlgorithm(dependencyGraph);
+		let importedPrograms: string = "";
+		importList.forEach((importFileName: string) => {
+			const currentImport = fs.readFileSync(defaultPath + path.sep + importFileName + ".pl");
+			importedPrograms = importedPrograms + currentImport;
+		});
+
+		return importedPrograms + program;
+}
+
+function getDependencyGraph(
+	program: string,
+	defaultPath: string,
+	currentFileName: string,
+	currentGraph?: Set<string[]>) {
+
+		if (currentGraph === undefined) {
+			currentGraph = new Set<string[]>();
+		}
+
+		regexGroupToArray(program, fetchPrologImportsRegExp, 1).forEach((fileName: string) => {
+			// Note: The ! in currentGraph! is there to tell this object will not be undefined (or null)
+			// See: https://www.typescriptlang.org/docs/handbook/release-notes/typescript-2-0.html (Non-null assertion operator)
```

Comment:
The non-null-assertion is unnecessary if you create a local variable `const graph = currentGraph;` right after the `undefined`-check in line 40 and use that variable in here instead.